doc/develop/languages/c/index.rst

```diff
@@ -12,6 +12,11 @@ available as part of the C header files under the :file:`include` directory, so
 writing Zephyr applications in C gives the developers access to the most
 features.
 
+The ``main()`` function must have the return type of ``int`` as Zephyr
+applications run in a "hosted" environment as defined by the C
+standard. Zephyr ignores the return value from main, so applications should
+not return status information and should, instead, return zero.
```

Comment:
Please describe the API policy for `main()` rather than describe current Zephyr implementation ("Zephyr ignores the return value of main").  Something like the statement below would work well.  Ideally this would be documented as a public API (even if the actual declaration was conditionally-compiled out):

- All non-zero return values from `main()` are reserved.
- All meanings of return values from `main()` are reserved.

Reply:
Yeah, you're right, the docs should state the policy not just the current implementation. Once that policy has been agreed upon, we can update the docs to match as part of this PR.

Reply:
I've changed the wording to require a return value of zero from main (at least for now).